### PR TITLE
Replace RabbitAutoConfiguration with local RabbitConfiguration

### LINKED ISSUE #187: Reconsider importing Boot's auto-configuration
RabbitServiceAutoConfiguration uses `@Import` to pull in Boot's `RabbitAutoConfiguration`: https://github.com/spring-cloud/spring-cloud-stream-binder-rabbit/blob/d4fbad5672140a9c5d3173df0e9a05bb58e7b994/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/config/RabbitServiceAutoConfiguration.java#L132

This has [caused some confusion](https://github.com/spring-projects/spring-boot/issues/14906) as it means that `@SpringBootApplication(exclude = {RabbitAutoConfiguration.class})` doesn't work. It also has the potential to cause ordering problems as other auto-configurations that have used `@AutoConfigureBefore(RabbitAutoConfiguration.class)` or `@AutoConfigureAfter(RabbitAutoConfiguration.class)` won't be ordered correctly.

More generally, other than when ordering auto-configurations using `@AutoConfigureBefore` or `@AutoConfigureAfter` or excluding them, I would recommend not depending on Boot's auto-configuration classes directly.

### PR DESCRIPTION
**Pull request by bono007**
Relies on https://github.com/spring-projects/spring-boot/pull/26982/files

@wilkinsona

Fixes #187

### CODE REVIEW

File: spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/config/RabbitConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+ * @author Chris Bono
+ * @since 3.2
+ */
+@Configuration(proxyBeanMethods = false)
+@EnableConfigurationProperties(RabbitProperties.class)
+public class RabbitConfiguration {
+
+	@Bean
+	@ConditionalOnMissingBean
+	RabbitConnectionFactoryBeanConfigurer rabbitConnectionFactoryBeanConfigurer(RabbitProperties properties,
```
Comment (bono007) [inline spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/config/RabbitConfiguration.java:57]: The binder context will skip these beans and instead re-use the beans from the SB RabbitAutoConfiguration when:
- there is no specific `spring.cloud.stream.<BINDER_NAME>.binder.environment.*` properties
- the SB RabbitAutoConfiguration is not excluded

File: spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/config/RabbitConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+		CachingConnectionFactory cachingConnectionFactory = new CachingConnectionFactory(connectionFactory);
+		rabbitCachingConnectionFactoryConfigurer.configure(cachingConnectionFactory);
+
+		return cachingConnectionFactory;
+	}
+
+	@Bean
+	@ConditionalOnSingleCandidate(ConnectionFactory.class)
+	@ConditionalOnMissingBean(RabbitOperations.class)
+	public RabbitTemplate rabbitTemplate(RabbitProperties rabbitProperties, ConnectionFactory connectionFactory, ObjectProvider<MessageConverter> messageConverter,
```
Comment (bono007) [inline spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/config/RabbitConfiguration.java:101]: **IGNORE FOR NOW** If the above approach is good for the connection factory, I will do something similar here for the RabbitTemplate.

Comment (bono007) [reply spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/config/RabbitConfiguration.java:101]: I will get back to this MR in the next 24-48 hrs to follow through w/ my comment above.

**Comment by bono007** [comment]
@garyrussell @sabbyanandan this one is ready for review.

**Comment by garyrussell** [comment]
LGTM; but shouldn't we rename `RabbitServiceAutoConfiguration` ?

**Comment by bono007** [comment]
Thx @garyrussell . I too agree we should rename it to `RabbitBinderConfiguration` as that is what it really is. This would be inline w/ its SCSK [counterpart](https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/blob/main/spring-cloud-stream-binder-kafka/src/main/resources/META-INF/spring.binders#L2). 

I just was not sure about putting it in w/ this current proposal.

**Comment by garyrussell** [comment]
Yes; it's ok to do it in this PR.

**Pull request merged**